test/e2e/operator.go

```diff
@@ -378,34 +385,108 @@ var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
 })
 
 var _ = Describe("ARO Operator - MUO Deployment", func() {
-	It("must be deployed by default with FIPS crypto mandated", func(ctx context.Context) {
-		muoIsDeployed := func() (bool, error) {
-			By("getting MUO pods")
-			pods, err := clients.Kubernetes.CoreV1().Pods("openshift-managed-upgrade-operator").List(ctx, metav1.ListOptions{
-				LabelSelector: "name=managed-upgrade-operator",
-			})
-			if err != nil {
-				return false, err
-			}
-			if len(pods.Items) != 1 {
-				return false, fmt.Errorf("%d managed-upgrade-operator pods found", len(pods.Items))
-			}
+	When("we have a fresh ARO cluster", func() {
+		It("must deploy MUO by default with FIPS crypto mandated", func() {
+			fn := hasMUOFIPSCryptoMandatedFn(context.Background())
 
-			By("getting logs from MUO")
-			b, err := clients.Kubernetes.CoreV1().Pods("openshift-managed-upgrade-operator").GetLogs(pods.Items[0].Name, &corev1.PodLogOptions{}).DoRaw(ctx)
-			if err != nil {
-				return false, err
-			}
+			err := wait.PollImmediate(interval, duration, fn)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
 
-			By("verifying that MUO has FIPS crypto mandated by reading logs")
-			return strings.Contains(string(b), `msg="FIPS crypto mandated: true"`), nil
-		}
+	When("MUO is deleted on the ARO cluster", func() {
+		It("should reconcile the MUO deployment", func() {
+			ctx := context.Background()
+			fn := doesMUODeploymentExistFn(ctx)
 
-		err := wait.PollImmediate(30*time.Second, 10*time.Minute, muoIsDeployed)
-		Expect(err).NotTo(HaveOccurred())
+			By("waiting for the MUO deployment to be ready")
+			err := wait.PollImmediate(interval, duration, fn)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("deleting the MUO deployment")
+			err = deleteMUODeployment(ctx)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("waiting for the MUO deployment to be reconciled")
+			err = wait.PollImmediate(interval, duration, fn)
+			Expect(err).NotTo(HaveOccurred())
+		})
 	})
 })
 
+func getMUOPods(ctx context.Context) (*corev1.PodList, error) {
```

Comment:
Similar comment to above - the convention in this file is to define necessary functions such as this inside of the test spec rather than outside of it (`ARO Operator - Azure Subnet Reconciler` is a good example of one that has multiple) so I think we should keep that convention here.

Reply:
Done.